packages/content-utils/src/runtime/git.ts

```diff
@@ -101,19 +114,77 @@ type TrackedCommits = {
  * @internal
  */
 export async function getAllTrackedCommitDates(): Promise<TrackedCommits> {
+	const repoRoot = getRepoRoot();
 	const trackedFiles = await listGitTrackedFiles();
 
+	const args = ['log', '--format=t:%ct', '--name-status', '--', contentPath];
+
+	const gitLog = spawnSync('git', args, {
+		cwd: repoRoot,
+		encoding: 'utf-8',
+	});
+
+	if (gitLog.error) {
+		return {
+			oldest: [],
+			latest: [],
+		};
+	}
+
+	const logLines = gitLog.stdout.trim().split('\n');
+
+	let runningDate: number = Date.now();
+	const runningMinMax = new Map<string, { min: number; max: number }>();
+
+	for (const logLine of logLines) {
+		if (logLine.startsWith('t:')) {
+			// t:<seconds since epoch>
+			runningDate = Number.parseInt(logLine.slice(2)) * 1000;
+		}
+
+		// - Added files take the format `A\t<file>`
+		// - Modified files take the format `M\t<file>`
+		// - Deleted files take the format `D\t<file>`
+		// - Renamed files take the format `R<count>\t<old>\t<new>`
+		// - Copied files take the format `C<count>\t<old>\t<new>`
+		// The name of the file as of the commit being processed is always
+		// the last part of the log line.
+		const tabSplit = logLine.lastIndexOf('\t');
+		if (tabSplit === -1) continue;
+		const fileName = logLine.slice(tabSplit + 1);
+
+		const minMax = runningMinMax.get(fileName);
+
+		if (minMax === undefined) {
+			runningMinMax.set(fileName, {
+				min: runningDate,
+				max: runningDate,
+			});
+			continue;
+		}
+
+		minMax.min = Math.min(minMax.min, runningDate);
+		minMax.max = Math.max(minMax.max, runningDate);
+	}
+
 	const result: TrackedCommits = {
 		oldest: [],
 		latest: [],
 	};
 
 	for (const file of trackedFiles) {
+		// git log returns file names relative to the repo root
+		// convert the tracked paths to match that format
+		const repoFilePath = relative(repoRoot, resolve(contentPath, file));
+
+		const minMax = runningMinMax.get(repoFilePath);
+		if (minMax === undefined) continue;
+
 		// Replace the first occurrence of the separator so it doesn't get partially normalized when mixin Windows and Unix-like systems.
 		const name = file.replace(sep, ':');
 
-		result.oldest.push([name, await getCommitDate(file, 'oldest')]);
```

Comment:
is the getCommitDate fn still used somewhere? otherwise may want to delete it

Reply:
It is used on the dev server, it only reads the date for files as they are rendered.